### PR TITLE
[MM-10511] Fix missing users' status indicators on more DM modal

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
Fix missing users' status indicators on more DM modal

#### Ticket Link
Jira ticket: [MM-10511](https://mattermost.atlassian.net/browse/MM-10511)

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Added or updated unit tests (required for all new features)

### CODE REVIEW

File: components/more_direct_channels/more_direct_channels.jsx [hunk 1]
```diff
@@ -282,7 +301,7 @@ export default class MoreDirectChannels extends React.Component {
     }
 
     handleSubmitImmediatelyOn = (value) => {
-        return value.id === this.props.currentUserId || value.delete_at;
+        return value.id === this.props.currentUserId || Boolean(value.delete_at);
```
Comment (saturninoabril) [inline components/more_direct_channels/more_direct_channels.jsx]: Not related to the issue but decided to fix while I'm adding test to this component.  The consumer of this function uses it for boolean flags and so returning an explicit boolean value is the best to do it.

**Comment by jespino** [comment]
@saturninoabril probably you may want to change the base branch to release-5.0 before the merge

**Comment by saturninoabril** [comment]
Right, done changing base branch to release-5.0

**Comment by cpanato** [comment]
merging to the release branch

**Pull request merged**